src/test/com/github/prasanthj/hll/TestHyperLogLog.java

```diff
@@ -223,4 +266,31 @@ public void testHLLSparseMoreRegisterBits() {
     double delta = threshold * size / 100;
     assertEquals((double) size, (double) hll.count(), delta);
   }
+
+  @Test
+  public void testHLLSquash() {
+    final int size = 1000;
+
+    HyperLogLog hlls[] = new HyperLogLog[16];
+    for (int k = 8; k < hlls.length; k++) {
+      final HyperLogLog hll = HyperLogLog.builder()
+          .setEncoding(EncodingType.DENSE).setNumRegisterIndexBits(k).build();
+      for (int i = 0; i < size; i++) {
+        hll.addLong(i);
+      }
+      hlls[k] = hll;
+    }
+
+    for (int k = 8; k < hlls.length; k++) {
+      for (int j = k + 1; j < hlls.length; j++) {
+        final HyperLogLog large = hlls[j];
+        final HyperLogLog small = hlls[k];
+        final HyperLogLog mush = large.squash(small.getNumRegisterIndexBits());
+        assertEquals(small.count(), mush.count(), 0);
+        double threshold = size > 40000 ? longRangeTolerance : shortRangeTolerance;
+        double delta = threshold * size / 100;
+        assertEquals((double) size, (double) mush.count(), delta);
+      }
+    }
+  }
```

Comment:
Can you also add a test for p=14 to p=10 merge with DENSE + DENSE merge and SPARSE + DENSE merge?